acme/dns_challenge.go

```diff
@@ -19,37 +20,49 @@ var preCheckDNS preCheckDNSFunc = checkDNS
 
 var preCheckDNSFallbackCount = 5
 
-// DNSProvider represents a service for managing DNS records.
-type DNSProvider interface {
-	CreateTXTRecord(fqdn, value string, ttl int) error
-	RemoveTXTRecord(fqdn, value string, ttl int) error
+// DNSRecord returns the DNS record which will fulfill a DNS challenge
+func DNSRecord(domain, keyAuth string) (fqdn string, value string, ttl int) {
```

Comment:
Oh yes, right... the TLS-SNI-01 function for the cert needs docs as well.

What I meant was it should be clear that client implementations need to call these functions in order to get the needed information for the challenge. I think people may not necessarily read the implementations of the challenges while trying to write their own.

I guess the `ChallengeProvider` might be the place to document this.